flink-core/src/main/java/org/apache/flink/util/FileUtils.java

```diff
@@ -258,7 +261,82 @@ public static boolean deletePathIfEmpty(FileSystem fileSystem, Path path) throws
 			return false;
 		}
 	}
-	
+
+	/**
+	 * Check whether the two given filesystem is the same or not
+	 *
+	 * @param srcFs
+	 * @param destFs
+	 * @return
+	 */
+	public static boolean compareFs(FileSystem srcFs, FileSystem destFs) {
+		URI srcUri = srcFs.getUri();
+		URI dstUri = destFs.getUri();
+
+		// check schema
+		if (srcUri.getScheme() == null) {
+			return false;
+		}
+		if (!srcUri.getScheme().equals(dstUri.getScheme())) {
+			return false;
+		}
+
+		// check ports
+		if (srcUri.getPort() != dstUri.getPort()) {
+			return false;
+		}
+
+		// check ip
+		String srcHost = srcUri.getHost();
+		String dstHost = dstUri.getHost();
+		if ((srcHost != null) && (dstHost != null)) {
+			if (!srcHost.equals(dstHost)) {
+				try {
+					srcHost = InetAddress.getByName(srcHost).getCanonicalHostName();
+					dstHost = InetAddress.getByName(dstHost).getCanonicalHostName();
```

Comment:
I think FQDN is better when multiple service running on the same machine, what do yo think?